habitat_baselines/rl/ddppo/policy/resnet_policy.py

```diff
@@ -257,6 +263,35 @@ def __init__(
             self.compass_embedding = nn.Linear(input_compass_dim, 32)
             rnn_input_size += 32
 
+        if ImageGoalSensor.cls_uuid in observation_space.spaces:
+            goal_observation_space = spaces.Dict(
+                {"rgb": observation_space.spaces[ImageGoalSensor.cls_uuid]}
+            )
+            self.goal_visual_encoder = ResNetEncoder(
+                goal_observation_space,
+                baseplanes=resnet_baseplanes,
+                ngroups=resnet_baseplanes // 2,
+                make_backbone=getattr(resnet, backbone),
+                normalize_visual_inputs=normalize_visual_inputs,
+                obs_transform=obs_transform,
+            )
+
+            self.goal_visual_fc = nn.Sequential(
+                Flatten(),
+                nn.Linear(
+                    np.prod(self.goal_visual_encoder.output_shape), hidden_size
+                ),
+                nn.ReLU(True),
+            )
+
+            rnn_input_size += hidden_size
+
+        if force_blind_policy:
+            # we need to keep RGB SENSOR in simulator for ImageGoalSensor
+            # So we remove it here if we want blind policy
+            logger.info(f"Removing sensors from observation space")
+            observation_space = spaces.Dict({})
```

Comment:
Maybe, cleaner way will be do that during ResNetEncoder constructor call instead of overwriting `observation_space` that includes non visual sensors: 
in line 298: ```
self.visual_encoder = ResNetEncoder(
            observation_space if not force_blind_policy else spaces.Dict({}),
```